entc/gen/type.go

```diff
@@ -484,10 +484,15 @@ func (t *Type) AddIndex(idx *load.Index) error {
 	}
 	for _, name := range idx.Fields {
 		f, ok := t.fields[name]
-		if !ok {
-			return fmt.Errorf("unknown index field %q", name)
+		if !ok || f == nil {
+			// Might be the ID field, so check
+			if t.ID.Name == name {
+				f = t.ID
+			} else {
+				return fmt.Errorf("unknown index field %q", name)
+			}
 		}
-		if f.def.Size != nil && *f.def.Size > schema.DefaultStringLen {
+		if f.def != nil && f.def.Size != nil && *f.def.Size > schema.DefaultStringLen {
 			return fmt.Errorf("field %q exceeds the index size limit (%d)", name, schema.DefaultStringLen)
```

Comment:
This one is not needed, just use `NewType` in the test. 

```suggestion
		if def.Size != nil && *f.def.Size > schema.DefaultStringLen {
			return fmt.Errorf("field %q exceeds the index size limit (%d)", name, schema.DefaultStringLen)
```

Reply:
Probably should be `f.def.Size` not `def.Size`.

Also, I added this in because when I added the change to use `t.ID` as `f`, `f.def` was `nil` and hence caused a panic.

Happy to remove it but from what I could tell, `def` is not assigned on the `ID` field.

Reply:
So removing that again actually brings back that error. This is from the PR action, but I can also repro locally.

![image](https://user-images.githubusercontent.com/8946502/100953179-23e3cc80-3566-11eb-8d44-911f07a30142.png)